src/components/mods/RandomHeldItemsMod.ts

```diff
@@ -0,0 +1,203 @@
+            {
+                "name": "Thundertone",
+                "probability": .005
+            }
+        ],
+        "Ground": [
+            {
+                "name": "Guard Spec",
+                "probability": .005
+            }
+        ],
+        "Psychic": [
+            {
+                "name": "Ultra Ball",
+                "probability": .005
+            }
+        ],
+        "Rock": [
+            {
+                "name": "Escape Rope",
+                "probability": .05
+            },
+            {
+                "name": "Iron",
+                "probability": .005
+            }
+        ],
+        "Ice": [
+            {
+                "name": "Ice Heal",
+                "probability": .025
+            },
+            {
+                "name": "TM13",
+                "probability": .005
+            }
+        ],
+        "Bug": [
+            {
+                "name": "Repel",
+                "probability": .025
+            },
+            {
+                "name": "Max Repel",
+                "probability": .005
+            }
+        ],
+        "Dragon": [
+            {
+                "name": "Rare Candy",
+                "probability": .0025
+            }
+        ],
+        "Ghost": [
+            {
+                "name": "Full Restore",
+                "probability": .0025
+            }
+        ]
+     };
+
+     /**
+      * Name of the mod.
+      */
+     public readonly name: string = "Random Held Items";
+
+     /**
+      * Mod events, keyed by name.
+      */
+     public readonly events: ICallbackRegister = {
+         [this.eventNames.onWildGrassPokemonChosen]: (chosenInfo: INewPokemon) => {
+             const pokemonName: string = chosenInfo.title.join("");
+             const pokemonType: string = this.gameStarter.constants.pokemon.byName[pokemonName].types[0];
+             const probabilityOfHeldItem: number = this.gameStarter.numberMaker.randomReal1();
+             const chosenItem = this.randomHeldItemGenerator(chosenInfo, probabilityOfHeldItem, pokemonType);
+
+             if (chosenItem[0] !== "") {
+                 chosenInfo.item = chosenItem;
+             }
+         }
+     };
+
+     /**
+      * Handles which item is chosen for onNewPokemonCreation.
+      *
+      * @param counter   Keeps track of item probabilities.
+      * @param chosenInfo   Info chosen by chooseRandomWildPokemon.
+      * @param chosenItem   Item chosen by the function, if none then equal to "".
+      * @param probabilityOfHeldItem   Randomly generated number for probability of the wild pokemon having an item.
+      * @param pokemonType   Type of the wild encountered Pokemon.
+      *
+      * @returns Returns the name of an item or "" if no item generated.
+      */
+     private randomHeldItemGenerator(chosenInfo: INewPokemon, probabilityOfHeldItem: number, pokemonType: string): string[] {
+            let counter: number = 0;
+            let chosenItem: string[] = [];
```

Comment:
Challenge: don't create a temporary `chosenItem` variable. Simpler functions are better, and you can simplify this function.